engine/src/main/java/org/camunda/bpm/engine/impl/cmd/DeployCmd.java

```diff
@@ -186,7 +188,8 @@ protected DeploymentWithDefinitions doExecute(final CommandContext commandContex
   }
 
   protected void acquireExclusiveLock(CommandContext commandContext) {
-    if (commandContext.getProcessEngineConfiguration().isDeploymentLockUsed()) {
+    if (commandContext.getProcessEngineConfiguration().isDeploymentLockUsed()
+        && !DatabaseUtil.checkDatabaseType(DbSqlSessionFactory.CRDB)) {
```

Comment:
This should be implemented in `DbSqlSession#lock` to have it in one place and not repeated for every usage of pessimistic locks.

Reply:
Refactored. Also added code comments to all retryable commands with explanations on why this command can be retried.

Reply:
Any particular reason you implemented this in `DbEntityManager#lock` instead of `DbSqlSession#lock` where we already do the check for H2?

Reply:
Just because we use the `PesistenceSession` interface, so it means that it can be replaced with another implementation. On the other hand, the `DbEntityManager` class is always the same.

Reply:
I'd prefer to have it in the `DbSqlSession` to be honest. It is very unlikely there will other persistence session implementations for CRDB in the future. In contrast, by having this concern in a single place we reduce the code complexity for anyone who reads this until then.

Reply:
Already done :)